resources/axon.rb

```diff
@@ -108,7 +108,7 @@
     mode '744' unless node['platform'] == 'windows'
   end
 
-  if node['platform'] != 'windows' && eg_install
+  if node['platform'] != 'windows' && (eg_driver_installer != nil || eg_service_installer != nil)
```

Comment:
Might be worthwhile to put all this logic into a method `should_install_eg?` or something. Then you can log extra warning messages in it if someone specifies incompatible options such as `eg_install=true` and `eg_driver_installer=nil`.

Reply:
eg_install at this point should only be for windows axon agents since it bundles the driver and the service with the agent installer. So at some point, we might have a flag to not install the EG service like the Java agents have. I'd rather decouple the eg_installer attribute from linux entirely since just leaving the EG driver and service attributes empty is sufficient. 

Moving to a method is a good idea though. I'm not sure its right for this commit at the moment but I'll make a note to move some of the redundant checks to their own methods.